### ISSUE TITLE
Removing hard reference to project Modulex and ModuleY throws exception at runtime

### ISSUE THREAD

**Issue by MonkeyJumps** [devkimchi/Autofac-Dynamic-Moduling#1]
In short, I am using this project to duplicate a specific issue I currently have with my personal project. I have a need to dynamically load an assembly from file. The interface is in a shared dll that has a hard reference but the concrete class is not referenced.

I have the same set up as this project. However, I do not have a hard reference to moduleX and moduleY and I get the following exception at runtime:

`{"The requested service 'ComponentA.IComponentA' has not been registered. To avoid this exception, either register a component to provide the service, check for service registration using IsRegistered(), or use the ResolveOptional() method to resolve an optional dependency."}`

In other words, your example works perfectly but what if I can't have a hard reference to my Module?
